api/envoy/config/filter/network/http_connection_manager/v2/http_connection_manager.proto

```diff
@@ -450,6 +452,42 @@ message HttpConnectionManager {
   // with `prefix` match set to `/dir`. Defaults to `false`. Note that slash merging is not part of
   // `HTTP spec <https://tools.ietf.org/html/rfc3986>` and is provided for convenience.
   bool merge_slashes = 33;
+
+  // Configuration of local reply send by envoy for http responses. Allows to define custom rules
+  // which maps response attribute returned by envoy to different one. Currently mapping of
+  // status code is implemented.
+  message SendLocalReplyConfig {
+    // Configuration which allows to define custom rules for matching local response.
+    // It works like and operator e.g: if you define status code 404 and body pattern
+    // then response must match both
+    message Match {
+      oneof matcher_specifier{
+        // Definition of status codes to match.
+        google.protobuf.UInt32Value status_codes = 1;
```

Comment:
Effectively I think this can be a header matcher on `:status`? Per @htuch point elsewhere, is there anything we can do here to share some of this beyond just sharing HeaderMatcher, StringMatcher, etc.?